main.go

```diff
@@ -14,23 +15,413 @@ import (
+// it unregister system from Red Hat Subscription Management
+func disconnectAction(ctx *cli.Context) error {
+	var start time.Time
+	durations := make(map[string]time.Duration)
+	errorMessages := make(map[string]error)
+	hostname, err := os.Hostname()
+	if err != nil {
+		return cli.Exit(err, 1)
+	}
+	fmt.Printf("Disconnecting %v from %v.\nThis might take a few seconds.\n\n", hostname, Provider)
+
+	_, disconnectedPrefix, errorPrefix, isColorful := getColorPreferences(ctx)
+
+	/* 1. Deactivate rhcd daemon */
+	start = time.Now()
+	progressMessage := fmt.Sprintf(" Deactivating the %v daemon", BrandName)
+	err = showProgress(progressMessage, isColorful, deactivate)
+	if err != nil {
+		errorMessages[BrandName] = fmt.Errorf("cannot deactivate daemon: %w", err)
+		fmt.Printf(errorPrefix+" Cannot deactivate the %v daemon\n", BrandName)
+	} else {
+		fmt.Printf(disconnectedPrefix+" Deactivated the %v daemon\n", BrandName)
+	}
+	durations[BrandName] = time.Since(start)
+
+	/* 2. Disconnect from Red Hat Insights */
+	start = time.Now()
+	err = showProgress(" Disconnecting from Red Hat Insights...", isColorful, unregisterInsights)
+	if err != nil {
+		errorMessages["insights"] = fmt.Errorf("cannot disconnect from Red Hat Insights: %w", err)
+		fmt.Printf(errorPrefix + " Cannot disconnect from Red Hat Insights\n")
+	} else {
+		fmt.Print(disconnectedPrefix + " Disconnected from Red Hat Insights\n")
+	}
+	durations["insights"] = time.Since(start)
+
+	/* 3. Unregister system from Red Hat Subscription Management */
+	err = showProgress(" Disconnecting from Red Hat Subscription Management...", isColorful, unregister)
+	if err != nil {
+		errorMessages["rhsm"] = fmt.Errorf("cannot disconnect from Red Hat Subscription Management: %w", err)
+		fmt.Printf(errorPrefix + " Cannot disconnect from Red Hat Subscription Management\n")
+	} else {
+		fmt.Printf(disconnectedPrefix + " Disconnected from Red Hat Subscription Management\n")
+	}
+	durations["rhsm"] = time.Since(start)
+
+	fmt.Printf("\nManage your Red Hat connector systems: https://red.ht/connector\n")
+
+	showTimeDuration(durations)
+
+	err = showErrorMessages("disconnect", errorMessages)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// canonicalFactAction tries to gather canonical facts about system,
+// and it prints JSON with facts to stdout.
+func canonicalFactAction(_ *cli.Context) error {
+	// NOTE: CLI context is not useful for anything
+	facts, err := GetCanonicalFacts()
+	if err != nil {
+		return cli.Exit(err, 1)
+	}
+	data, err := json.MarshalIndent(facts, "", "   ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(data))
+	return nil
+}
+
+// statusAction tries to print status of system. It means that it gives
+// answer on following questions:
+// 1. Is system registered to Red Hat Subscription Management?
+// 2. Is system connected to red Hat Insights?
+// 3. Is rhsm.service running?
```

Comment:
I think this is a mental slip-up, but this should be rhcd.service, I think?

Reply:
Yeah, you are right. I'm sorry.